ui/prometheus-plugin/src/model/prometheus-client.ts

```diff
@@ -65,21 +65,23 @@ function useQueryWithGet<T extends RequestParams<T>, TResponse>(
   params: T,
   queryOptions?: QueryOptions
 ) {
-  const config = usePrometheusConfig(queryOptions?.datasource);
-  const datasourceURL = buildDatasourceURL(config.metadata.name, config.spec.http);
-  const key = [datasourceURL, apiURI, params] as const;
+  const getPrometheusDatasource = useGetPrometheusConfig();
```

Comment:
a bit overkill to create an alias of the function don't you think ?

Reply:
Good question, not sure if you're asking:

1. Shouldn't this line just be `const getPrometheusConfig = useGetPrometheusConfig()` or
2. Why not just call `useGetPrometheusConfig()` to get the config instead of a callback to get it?

The answer to number 2 is a little more complicated and has to do with the rules of hooks in React, so happy to explain if it's the latter. If it's number 1, yeah, probably should just be the same name. 😂

Reply:
ah I was more on the number 2.
But if it's actually a necessary line of code and not just there to rename the function, then maybe just give the same name no ?